src/Parser/RestoDruid/Modules/Features/StatWeights.js

```diff
@@ -0,0 +1,316 @@
+ * matter because it *would've* just caused the next heal to overheal", but how far can we take this logic? My approach is that
+ * it doesn't only matter how much is healed, but also how quickly. We'll count all healing that doesn't overheal even if we
+ * theorize that it might lead to overhealing. We're more strict on heals that overheal: they're disregarded entirely even
+ * if they're only partial overhealing. This is because a spell that partially overheals will do the same amount of effective
+ * healing regardless of the raw strength of the heal. This approach can cause some possible weirdness. For example, consider
+ * a situation where there is a heal for 500 (0 overheal) followed immediately by a heal for 400 (+ 100 overheal). 500 of this
+ * healing would be counted towards stat weights. Now consider a situation where one heal does 900 (+ 100 overheal). Functionally,
+ * this situation is identical, but in this case none of the healing is counted towards stat weights. Counting all effective
+ * healing, even heals that partially overheal, fixes this issue and is overall a valid approach, but still not one I will be taking.
+ * I think an advantage of disregarding all partial overheals is that it gives "top off" heals an effective lower weight than
+ * "life saver" heals. Still, this is a decision I will revisit.
+ *
+ * Stat Tracking -
+ * These calculations work best when they use the players actual stats at the moment of a heal. As such, accuracy will be
+ * improved when a StatTracker module is implemented, rather than being forced to use the player's stats at the moment of pull.
+ * For now, I'm handling the most common big Druid stat buffs manually.
+ *
+ * Intellect -
+ * Math here is straightforward, as spells that scale with int scale directly and linearly with total int. Due to the 'all leather'
+ * bonus, each point of int gained is multiplied by 1.05, which has to be taken into account. The increase in power from
+ * one int is simply 1.05 / totalInt.
+ *
+ * Crit -
+ * The general formula is critChanceFromOneRating * healAmount, but we have to make sure to account for anything that makes a
+ * crit stronger than 2x (e.g. Drape of Shame), and we also have to account for if healAmount is already a crit. If the heal
+ * is already a crit then we double count the weight, so we first have to normalize it to the amount it would have done
+ * had it not crit. Special handling is needed for Living Seed obviously. We assume that higher crit will result in proportionally
+ * more Living Seeds, and so get the bonus to Living Seed as critChangeFromOneRating / critChance.
+ *
+ * Haste -
+ * Calculating the benefit to healing from being able to cast faster is a huge pain in the ass, particularly for Resto Druids
+ * who aren't very dependant on cast times. However, HoTs tick faster and effectively do increased healing per cast due to Haste.
+ * We make the decision to only calculate this benefit, which is fairly simple math:
```

Comment:
Doesn't this result in a very large undervaluation of Haste? The increased cast speed is an important part of the gain and for many other specs reason enough to use gear with Haste on it. Imagine if Resto Druid had 0 gain from HoTs ticking faster; would that mean Resto Druids would aim for 0 Haste? Your screenshot reveals Haste when just valuing HoT tick rate gain to already be close to Int, I reckon if you included cast speed (and thus reduced GCD) it would go much beyond.

I'm surprised by this approach and think it might be misleading. Is this the regular way to calculate the Haste value within the Resto Druid TC community?

Reply:
It does certainly undervalue Haste. This is a common calculation in the Resto Druid TC community, and we just say it's undervalued by ¯\_(ツ)_/¯

I could be mistaken, but I believe Resto Druid is less interested in faster cast speed than other healing classes. Our main heal is instant, and we do not often need to spam heals. Yes I know Haste reduces GCD for an instant spell, but unlike for a cast time spell Haste doesn't reduce the time in between when you decide to cast and when it lands.

If I can think of a good approach, I'll try to re-add Haste HPCT. I'm just not sure how best to do it.

Reply:
We discussed this on Discord some time back and I concluded it's a major challenge when doing log-based stat weight calculation. 🤷‍♂️ I was thinking of a super simplified approach; 1% Haste = 1% more healing from all spells affected by Haste (GCD or channel), the beauty lies in that it's an extremely simple approach that is right in a way.